tools/clientSimulator/pfClientSimulator.py

```diff
@@ -175,34 +185,33 @@ def main():
                 testLauncher,
                 testFactory.generateTestVector()).launchInteractiveMode()
         else:
-            while True:
-                scenarioOptions = {
-                    scenarioNumber:
-                        (scenarioFileName,
-                         DynamicCallHelper(
-                             launchScenario,
-                             logger,
-                             consoleLogger,
-                             configParser["ActionGathererFile"],
-                             os.path.join(
-                                 configParser["ScenariosDirectory"], scenarioFileName),
-                             testFactory,
-                             testLauncher
-                         ))
-                    for scenarioNumber, scenarioFileName in enumerate(
-                        [file for file in sorted(os.listdir(
-                            configParser["ScenariosDirectory"]))])
-                }
-                if args.scenario is not None:
-                    scenarioOptions[args.scenario][1]()
-                    # Let the user choose other scenario after the one choosed by command line
-                    args.scenario = None
-                else:
-                    UserInteractor.getMenu(scenarioOptions)
+            scenarioOptions = [
+                    (scenarioFileName,
+                     DynamicCallHelper(
+                         launchScenario,
+                         logger,
+                         consoleLogger,
+                         configParser["ActionGathererFile"],
+                         os.path.join(
+                             configParser["ScenariosDirectory"], scenarioFileName),
+                         testFactory,
+                         testLauncher
+                     ))
+                for scenarioFileName in sorted(os.listdir(configParser["ScenariosDirectory"]))
+            ]
+            if args.scenario is not None:
+                for elem in args.scenario:
+                    scenarioOptions[elem][1]()
+            if args.no_exit:
+                # Let the user choose more scenarios after the ones chosen by command line
+                # or if none was given on the command line.
+                UserInteractor.getMenu(scenarioOptions, "Quit")
```

Comment:
In this actual version, if the user wants to get to the menu, no_exit option must be put on the command line, which does not seem natural for me...